### PR TITLE
Update number of shells in SimulationState

### PR DESCRIPTION
**Pull request by DeerWhale**
### :pencil: Description

**Type:** :beetle: `bugfix` | :vertical_traffic_light: `testing` 

Update the `no_of_shells` attribute in SimulationState to be the `no_of_shells_active` in geometry. 
This can make sure the dimensionality of the attributes in the SimulationState is consistent with the activate shells after changing the v_inner_boundary value on fly. 

### :vertical_traffic_light: Testing

How did you test these changes?

- [x] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Comment by andrewfullard** [comment]
These changes in model/base.py may be replaced by work @wkerzendorf is doing. Or this should be merged into #2445

File: tardis/model/tests/test_csvy_model.py [hunk 1]
```diff
@@ -75,6 +76,26 @@ def test_compare_models(model_config_fnames):
     )
 
 
+def test_dimensionality_after_update_v_inner_boundary(
+    model_config_fnames,
```
Comment (Knights-Templars) [inline tardis/model/tests/test_csvy_model.py:80]: See if there are fixtures that loads the model.

File: tardis/model/tests/test_csvy_model.py [hunk 2]
```diff
@@ -75,6 +76,26 @@ def test_compare_models(model_config_fnames):
...
+
+    assert new_csvy_model.no_of_raw_shells == csvy_model.no_of_raw_shells
+    assert new_csvy_model.no_of_shells == csvy_model.no_of_shells - 1
+    assert new_csvy_model.velocity.shape[0] == csvy_model.velocity.shape[0] - 1
+    assert new_csvy_model.density.shape[0] == csvy_model.density.shape[0] - 1
+    assert new_csvy_model.volume.shape[0] == csvy_model.volume.shape[0] - 1
+
+
 @pytest.fixture(scope="module")
 def csvy_model_test_abundances(example_csvy_file_dir):
```
Comment (sonachitchyan) [inline tardis/model/tests/test_csvy_model.py:100]: this fixture can be maybe good to use in the test above?

File: tardis/model/tests/test_csvy_model.py (hunk 1 above)
Comment (jvshields) [reply tardis/model/tests/test_csvy_model.py:80]: If there's isn't already a fixture for this, we should write one. In general I think unless there's a very good reason for it, no test should reload these generic objects or the full list of tests will reload things many times without much point.

File: tardis/model/tests/test_csvy_model.py (hunk 2 above)
Comment (DeerWhale) [reply tardis/model/tests/test_csvy_model.py:100]: created new fixture in this test.py

File: tardis/model/tests/test_csvy_model.py (hunk 1 above)
Comment (DeerWhale) [reply tardis/model/tests/test_csvy_model.py:80]: created new fixture in this test.py

**Comment by wkerzendorf** [comment]
@DeerWhale the tests fail.

**Comment by Rodot-** [comment]
Is there any update on this?  I believe CSVY models are still broken until this is merged.

**Comment by DeerWhale** [comment]
Closing this duplicated one -- the (almost) same edits is made in a new PR #2504 after the restructure.
